src/main/java/com/amazon/opendistroforelasticsearch/security/compliance/ComplianceConfig.java

```diff
@@ -45,205 +45,315 @@
 import org.apache.logging.log4j.Logger;
 import org.elasticsearch.ElasticsearchException;
 import org.elasticsearch.common.settings.Settings;
-import org.elasticsearch.env.Environment;
 import org.joda.time.DateTime;
 import org.joda.time.DateTimeZone;
 import org.joda.time.format.DateTimeFormat;
 import org.joda.time.format.DateTimeFormatter;
 
-import com.amazon.opendistroforelasticsearch.security.auditlog.AuditLog;
-import com.amazon.opendistroforelasticsearch.security.resolver.IndexResolverReplacer;
-import com.amazon.opendistroforelasticsearch.security.resolver.IndexResolverReplacer.Resolved;
 import com.amazon.opendistroforelasticsearch.security.support.ConfigConstants;
 import com.amazon.opendistroforelasticsearch.security.support.WildcardMatcher;
 import com.google.common.cache.CacheBuilder;
 import com.google.common.cache.CacheLoader;
 import com.google.common.cache.LoadingCache;
+import com.google.common.collect.ImmutableSet;
 
-
+/**
+ * This class represents all configurations for compliance.
+ * DLS/FLS uses this configuration for filtering and anonymizing fields.
+ * Audit Logger uses this configuration to post compliance audit logs.
+ */
 public class ComplianceConfig {
 
-    private final Logger log = LogManager.getLogger(getClass());
-    private final Settings settings;
-	private final Map<String, Set<String>> readEnabledFields = new HashMap<>(100);
-    private final List<String> watchedWriteIndices;
-    private DateTimeFormatter auditLogPattern = null;
-    private String auditLogIndex = null;
-    private final boolean logDiffsForWrite;
-    private final boolean logWriteMetadataOnly;
-    private final boolean logReadMetadataOnly;
+    private static final Logger log = LogManager.getLogger(ComplianceConfig.class);
+    private static final int SALT_SIZE = 16;
+    private static final int CACHE_SIZE = 1000;
+    private static final String INTERNAL_ELASTICSEARCH = "internal_elasticsearch";
+
     private final boolean logExternalConfig;
     private final boolean logInternalConfig;
-    private final LoadingCache<String, Set<String>> cache;
+    private final boolean logReadMetadataOnly;
+    private final boolean logWriteMetadataOnly;
+    private final boolean logDiffsForWrite;
+    private final List<String> watchedWriteIndices;
```

Comment:
Should it be `Set<String> watchedWriteIndicesPatterns` to be consistent with `Set<String> immutableIndicesPatterns`?